data_steward/analytics/cross_ct_rt/CDR_QC_RT_vs_CT.py

```diff
@@ -0,0 +1,497 @@
+    FROM `{project_id}.{rt_dataset}.INFORMATION_SCHEMA.COLUMNS`
+    where lower(column_name) = 'person_id'
+);
+
+CREATE TEMP TABLE rt_person_id(person_id INT64);
+CREATE TEMP TABLE ct_person_id(person_id INT64);
+
+LOOP
+  SET i = i + 1;
+  IF i > ARRAY_LENGTH(person_tables) THEN
+    LEAVE;
+  END IF;
+  EXECUTE IMMEDIATE '''
+INSERT rt_person_id
+SELECT distinct person_id
+FROM `{project_id}.{rt_dataset}.''' || person_tables[ORDINAL(i)] || '''`
+''';
+
+END LOOP;
+
+set i = 0;
+LOOP
+  SET i = i + 1;
+  IF i > ARRAY_LENGTH(person_tables) THEN
+    LEAVE;
+  END IF;
+  EXECUTE IMMEDIATE '''
+INSERT ct_person_id
+SELECT distinct person_id
+FROM `{project_id}.{ct_dataset}.''' || person_tables[ORDINAL(i)] || '''`
+''';
+
+END LOOP;
+
+SELECT distinct count(*) as count_not_pass,
+  CASE WHEN count(*) > 0 THEN 'FAILURE' ELSE 'ok' END as success
+FROM rt_person_id
+WHERE person_id not in (select person_id from ct_person_id);
+"""
+
+df1=pd.read_gbq(query, dialect='standard') 
+
+
+if df1['count_not_pass'].sum()==0:
+ df = df.append({'query' : 'Query6 Verify All person_ids in RT exist in CT', 'result' : 'PASS'},  
+                ignore_index = True) 
+else:
+ df = df.append({'query' : 'Query6 Verify All person_ids in RT exist in CT' , 'result' : ''},  
+                ignore_index = True) 
+
+df1
+# -
+
+# # Verify All person_ids in CT includes expected RT drops
+# Ensure everyone dropped in RT for max age is as old or older than
+# the current threshold age of 'maximum_age' (89 years old).  If someone is dropped
+# from the RT and they are not removed by the age requirement,
+# something may be wrong.
+
+
+# +
+query = f"""
+DECLARE person_tables Array<STRING>;
+DECLARE i INT64 DEFAULT 0;
+
+set person_tables = Array(
+    SELECT table_name                                                                                                                                                                                
+    FROM `{project_id}.{rt_dataset}.INFORMATION_SCHEMA.COLUMNS`
+    where lower(column_name) = 'person_id'
+);
+
+CREATE TEMP TABLE rt_person_id(person_id INT64);
+CREATE TEMP TABLE ct_person_id(person_id INT64);
+
+LOOP
+  SET i = i + 1;
+  IF i > ARRAY_LENGTH(person_tables) THEN
+    LEAVE;
+  END IF;
+  EXECUTE IMMEDIATE '''
+INSERT rt_person_id
+SELECT distinct person_id
+FROM `{project_id}.{rt_dataset}.''' || person_tables[ORDINAL(i)] || '''`
+''';
+
+END LOOP;
+
+set i = 0;
+LOOP
+  SET i = i + 1;
+  IF i > ARRAY_LENGTH(person_tables) THEN
+    LEAVE;
+  END IF;
+  EXECUTE IMMEDIATE '''
+INSERT ct_person_id
+SELECT distinct person_id
+FROM `{project_id}.{ct_dataset}.''' || person_tables[ORDINAL(i)] || '''`
+''';
+
+END LOOP;
+
+SELECT distinct person_id,
+(2021 - year_of_birth) as age,
+CASE WHEN (2021 - year_of_birth) < {maximum_age} THEN 1 ELSE 0 END as Failure
```

Comment:
One more change: Can this extract the year from current date/datetime?

Reply:
@nishanthpp93 good catch!   It should use the current year and programmatically extract it.

Reply:
will do. thanks.